server/helpers/requests.ts

```diff
@@ -28,7 +31,21 @@ type PeerTubeRequestOptions = {
   jsonResponse?: boolean
 } & Pick<GotOptions, 'headers' | 'json' | 'method' | 'searchParams'>
 
-const peertubeGot = got.extend({
+
+let peertubeGot: Got
+if (proxy()) {
+  console.log('With Proxy Address')
+peertubeGot = got.extend({
+  agent: {
```

Comment:
Don't use an `if/else`: you duplicate too much code

Instead, inject the agent property: 

```

...getAgentIfNeeded(),

headers: {
...
```

and 

```
function getAgentIfNeeded () {
  if (!isProxyEnabled()) return {}

  return { 
    agent: {
      ....
    }
  }
}
```

Reply:
We have removed the redundancy and made the code smaller as you suggested.